lib/ansible/modules/packaging/os/redhat_subscription.py

```diff
@@ -503,14 +505,22 @@ def main():
     consumer_id = module.params["consumer_id"]
     force_register = module.params["force_register"]
 
+    global SUBMAN_CMD
+    SUBMAN_CMD = module.get_bin_path('subscription-manager', False)
```

Comment:
If you do
`module.get_bin_path('subscription-manager', True)`
then `get_bin_path` will call `self.fail_json` directly, do you can delete the lines
```
if not SUBMAN_CMD:
    module.fail_json(msg="Could not find subscription-manager. Please ensure it is installed.")
```